trinity/plugins/builtin/ethstats/ethstats_service.py

```diff
@@ -0,0 +1,86 @@
+import asyncio
+
+import websockets
+
+from p2p.service import (
+    BaseService
+)
+
+from trinity.plugins.builtin.ethstats.ethstats_client import (
+    EthstatsClient,
+)
+from eth.chains.base import (
+    BaseChain
+)
+from p2p.peer import (
+    PeerPool
+)
+
+
+class EthstatsService(BaseService):
+    def __init__(
+        self,
+        server_url: str,
+        server_secret: str,
+        chain: BaseChain,
+        peer_pool: PeerPool,
+        *args,
+        **kwargs
+    ) -> None:
+        super(EthstatsService, self).__init__(*args, **kwargs)
+
+        self.server_url: str = server_url
+        self.server_secret: str = server_secret
+
+        self.chain: BaseChain = chain
+        self.peer_pool: PeerPool = peer_pool
+
+    async def _run(self) -> None:
+        await self.connection_loop()
```

Comment:
minor but maybe just drop the `connection_loop` function in favor of putting the logic here in the `_run` method?

Reply:
Moved `connection_loop` into `_run`.